Modules/_asynciomodule.c

```diff
@@ -40,12 +39,17 @@ typedef enum {
     PyObject *prefix##_source_tb;                                           \
     PyObject *prefix##_cancel_msg;                                          \
     PyObject *prefix##_cancelled_exc;                                       \
+    PyObject *prefix##_awaited_by;                                          \
     fut_state prefix##_state;                                               \
-    /* These bitfields need to be at the end of the struct
-       so that these and bitfields from TaskObj are contiguous.
+    /* Used by profilers to make traversing the stack from an external      \
+       process faster. */                                                   \
+    char prefix##_is_task;                                                  \
+    char prefix##_awaited_by_is_set;                                        \
```

Comment:
is this necessary to add, can't profilers check the type of `_awaited_by` if it is set or not?

Reply:
They can't, unfortunately. They only can access pointers in memory, they have no idea what the type of some random memory layout object is. In theory that information is obtainable but it will cost a lot. And with external profiling it's important to capture a snapshot as fast as possible. So while I agree, this seems to be inelegant, it's there for a reason.